txpipe/twopoint.py

```diff
@@ -322,7 +323,10 @@ def get_m(self, data, i):
         mask = (data['source_bin'] == i)
         
         # We use S=0 here because we have already included it in R_total
-        g1, g2 = apply_metacal_response(data['R_total'][i], 0.0, data['mcal_g1'][mask],data['mcal_g2'][mask])
+        if self.config['shear_catalog_type']=='metacal':
+            g1, g2 = apply_metacal_response(data['R_total'][i], 0.0, data['mcal_g1'][mask],data['mcal_g2'][mask])
+        elif self.config['shear_catalog_type']=='nonmetacal':
+            g1, g2 = apply_lensfit_calibration(data['mcal_g1'][mask],data['mcal_g2'][mask])
 
```

Comment:
Can you add an else here checking for unknown calibrators / wrong choice of the parameter.